Sourcery/Generating/Template/Swift/SwiftTemplate.swift

```diff
@@ -157,11 +164,85 @@ class SwiftTemplate: Template {
             throw SwiftTemplateParsingError.compilationError(path: binaryFile, error: compilationResult.error)
         }
 
-        let result = try Process.runCommand(path: binaryFile.description,
-                                            arguments: [serializedContextPath.description])
+        return binaryFile
+    }
+
+}
+
+class SwiftTemplate: NSObject, NSCoding, Template {
+
+    var sourcePath: Path
+    let code: String
+    let contentSha: String?
+
+    private lazy var buildDir: Path = Path.cleanTemporaryDir(name: "build")
 
+    private(set) var binaryPath: Path!
+
+    init(code: String, sourcePath: Path) {
+        self.code = code
+        self.sourcePath = sourcePath
+        self.contentSha = code.sha256()
+    }
+
+    init(path: Path, cachePath: Path?) throws {
```

Comment:
For me it feels that it's better to keep things separate, parsing is parsing, rendering is rendering. But I'll keep looking in how to organise code better, that's not the final version.